lib/contestant.rb

```diff
@@ -0,0 +1,30 @@
+require_relative './game'
+
+class Contestant < Game
+
+  attr_reader :first_name, :last_name, :age, :state_of_residence, :spending_money, :full_name, :game_interests
+
+  def initialize(info_hash)
+    @first_name = info_hash[:first_name]
+    @last_name = info_hash[:last_name]
+    @full_name = "#{@first_name} #{@last_name}"
+    @age = info_hash[:age]
+    @state_of_residence = info_hash[:state_of_residence]
+    @spending_money = info_hash[:spending_money]
+    @game_interests = []
+  end
+
+  def out_of_state?
+    @state_of_residence != 'CO'
+  end
+
+  def add_game_interest(name)
+    national_status = false
+    cost = 5 if name == 'Mega Millions'
+    cost = 2 if name == "Pick 4"
+    national_status = true if name == "Mega Millions"
+    current_interest = Game.new(name, cost, national_status)
+    @game_interests << current_interest.name
```

Comment:
this doesn't really make sense here-- you start with a name and then do a lot of work to build up a game object (but hardcoding, which again means that it won't work as soon as we introduce a new type of game to the lottery) and then at the end, you take that game object and just grab the name from it... which is exactly what you had to start from. so you turned a name into a game just to turn it back to a name, and you're not doing anything else with it-- this really only needs to be:

```
  def add_game_interest(game_name)
    @game_interests << game_name
  end
```